frontend/src/store/modules/shoots/getters.js

```diff
@@ -0,0 +1,247 @@
+  }
+}
+
+const getSortVal = (rootGetters, item, sortBy) => {
+  const value = getRawVal(rootGetters, item, sortBy)
+  const status = item.status
+  switch (sortBy) {
+    case 'purpose':
+      switch (value) {
+        case 'infrastructure':
+          return 0
+        case 'production':
+          return 1
+        case 'development':
+          return 2
+        case 'evaluation':
+          return 3
+        default:
+          return 4
+      }
+    case 'lastOperation': {
+      const operation = value || {}
+      const inProgress = operation.progress !== 100 && operation.state !== 'Failed' && !!operation.progress
+      const lastErrors = get(item, 'status.lastErrors', [])
+      const isError = operation.state === 'Failed' || lastErrors.length
+      const allErrorCodes = errorCodesFromArray(lastErrors)
+      const userError = isUserError(allErrorCodes)
+      const ignoredFromReconciliation = isReconciliationDeactivated(get(item, 'metadata', {}))
+
+      if (ignoredFromReconciliation) {
+        if (isError) {
+          return 400
+        } else {
+          return 450
+        }
+      } else if (userError && !inProgress) {
+        return 200
+      } else if (userError && inProgress) {
+        const progress = padStart(operation.progress, 2, '0')
+        return `3${progress}`
+      } else if (isError && !inProgress) {
+        return 0
+      } else if (isError && inProgress) {
+        const progress = padStart(operation.progress, 2, '0')
+        return `1${progress}`
+      } else if (inProgress) {
+        const progress = padStart(operation.progress, 2, '0')
+        return `6${progress}`
+      } else if (isShootStatusHibernated(status)) {
+        return 500
+      }
+      return 700
+    }
+    case 'readiness': {
+      const errorConditions = filter(get(status, 'conditions'), condition => get(condition, 'status') !== 'True')
+      const lastErrorTransitionTime = head(orderBy(map(errorConditions, 'lastTransitionTime')))
+      return lastErrorTransitionTime
+    }
+    case 'ticket': {
+      const { namespace, name } = item.metadata
+      return rootGetters.latestUpdatedTicketByNameAndNamespace({ namespace, name })
+    }
+    default:
+      return toLower(value)
+  }
+}
+
+export default {
+  filteredItems (state) {
+    return state.filteredShoots
+  },
+  itemByNameAndNamespace (state) {
+    return ({ namespace, name }) => {
+      return findItem(state)({ name, namespace })
+    }
+  },
+  selectedItem (state) {
+    if (state.selection) {
+      return findItem(state)(state.selection)
+    }
+  },
+  getShootListFilters (state) {
+    return state.shootListFilters
+  },
+  onlyShootsWithIssues (state, getters) {
+    return get(getters.getShootListFilters, 'onlyShootsWithIssues', true)
+  },
+  newShootResource (state) {
+    return state.newShootResource
+  },
+  initialNewShootResource (state) {
+    return state.initialNewShootResource
+  },
+  searchShoots (state, getters, rootState, rootGetters) {
```

Comment:
I think we should use a generic naming convention inside a module or class instead of repeating the name of the module or class in each and every method or property. We already do this e.g in `filteredItems`, `itemByNameAndNamespace` and `selectedItem`, `ITEM_PUT`, `HANDLE_EVENTS`, ... Each store module should have a similar structure and a generic naming convention. You can see this concept in the structure of a kubernetes resource with `kind`, `metadata`, `spec`, `items`, `status` properties. My proposal has been to name these new getter functions `searchItems` and `sortItems`.

Reply:
ok